src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -469,22 +478,34 @@ private static void AddDefaultVNetResourceNode(
      */
     public static String uploadCustomScript(final AzureVMAgentTemplate template, final String targetScriptName, TokenCache tokenCache) throws Exception {
         String targetStorageAccount = template.getStorageAccountName();
+        String targetStorageAccountType = template.getStorageAccountType();
         String resourceGroupName = template.getResourceGroupName();
         String location = template.getLocation();
         final Azure azureClient = tokenCache.getAzureClient();
 
         //make sure the resource group and storage account exist
         createAzureResourceGroup(azureClient, location, resourceGroupName);
         try {
-            createStorageAccount(azureClient, targetStorageAccount, location, resourceGroupName);
+            createStorageAccount(azureClient, targetStorageAccountType, targetStorageAccount, location, resourceGroupName);
         } catch (Exception e) {
             LOGGER.log(Level.INFO, e.getMessage());
         }
         CloudBlobContainer container = getCloudBlobContainer(azureClient, resourceGroupName, targetStorageAccount, Constants.CONFIG_CONTAINER_NAME);
         container.createIfNotExists();
-        CloudBlockBlob blob = container.getBlockBlobReference(targetScriptName);
+        CloudPageBlob blob = container.getPageBlobReference(targetScriptName);
         String scriptText = template.getInitScript();
-        blob.uploadText(scriptText, "UTF-8", AccessCondition.generateEmptyCondition(), null, null);
+        int stringSize = scriptText.length() % 512 == 0 ? scriptText.length() : scriptText.length() - scriptText.length() % 512 + 512;
```

Comment:
I would move this logic in its own function, or at least some comments are needed